Source generator fails when used in Visual Studio preview

## Describe the Bug

If the compiler in use doesn't match the source generator dependencies, code generation fails.

## Steps to Reproduce

Install VS preview, install Avatar nuget on a project. Compiler issues a warning like:

```
CSC : warning CS8785: Generator 'AvatarGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'TypeInitializationException' with message 'The type initializer for 'Avatars.WorkspaceServices' threw an exception.'
```

This later fails at run-time because the expected avatar types aren't present in the assembly.

## Expected Behavior

Things should Just Work :)

--- a/src/Avatar.Package/Avatar.Package.msbuildproj
+++ b/src/Avatar.Package/Avatar.Package.msbuildproj
@@ -22,6 +22,7 @@ service.AddBehavior(...);
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" Visible="false" Pack="false" />
+    <PackageReference Include="Microsoft.Net.Compilers.Toolset" Version="3.8.0" Visible="false" />
     <PackageReference Include="NuGetizer" Version="0.6.0" Visible="false" />
     <PackageReference Update="NuGetizer" Version="42.42.42" Visible="false" Condition="Exists('$(MSBuildThisFileDirectory)..\..\..\nugetizer\bin\')" />
   </ItemGroup>

--- a/src/Pack/Pack.props
+++ b/src/Pack/Pack.props
@@ -2,7 +2,7 @@
 
   <ItemGroup>
     <ProjectCapability Include="Pack" />
-    <ProjectReference Include="..\**\*.csproj" Exclude="..\**\bin\**\*.*;..\**\obj\**\*.*" />
+    <ProjectReference Include="..\**\*.csproj" Exclude="..\**\bin\**\*.*;..\**\obj\**\*.*;..\**\*Tests.csproj;..\Acceptance\**\*.csproj" />
   </ItemGroup>
   
   <!-- Remove IsPackable=false projects from the globing -->

